crates/fuel-streams-core/src/stream.rs

```diff
@@ -106,27 +104,36 @@ impl<S: Streamable> Stream<S> {
         &self,
         subject: &impl IntoSubject,
         payload: &S,
-    ) -> Result<u64, StreamError> {
+    ) -> Result<i64, StreamError> {
         let subject_name = &subject.parse();
-        self.store
-            .put(subject_name, payload.encode(subject_name).await.into())
-            .await
-            .map_err(|s| StreamError::PublishFailed {
+        let result = self
+            .store
+            .create(subject_name, payload.encode(subject_name).await.into())
+            .await;
+
+        match result {
+            Ok(sequence) => Ok(sequence as i64),
+            Err(e) if e.kind() == CreateErrorKind::AlreadyExists => {
+                // This is a workaround to avoid publish two times the same message
+                Ok(-1)
```

Comment:
```suggestion
            Ok(sequence) => Ok(()),
            Err(e) if e.kind() == CreateErrorKind::AlreadyExists => {
                // This is a workaround to avoid publish two times the same message
                Ok(())
```

Reply:
good point, done. Ty.